Src/FluentAssertions/Equivalency/Steps/EqualityComparerEquivalencyStep.cs

```diff
@@ -21,6 +21,14 @@ public EqualityComparerEquivalencyStep(IEqualityComparer<T> comparer)
             return EquivalencyResult.ContinueWithNext;
         }
 
+        if (comparands.Subject is null || comparands.Expectation is null)
+        {
+            // The later check for `comparands.Subject is T` leads to an failure even if the expectation is null
+            // IEqualityComparer isn't null aware. To preserve NullReferenceExceptions a compare with null is
+            // not handled by the custom comparer.
```

Comment:
❓ I don't get the remark about NREs. Isn't this change just going to make sure that the entire step (and the custom implementation of `IEqualityComparer<T>` is not going to be used) when either the subject or expectation is `null`?

Reply:
You are right. I removed this part of the comment. I'm not sure if the commet at all is usefull. Should I remove it?